pepper-apis/docs/specification/src/endpoints/studies.invitation-check.yml

```diff
@@ -0,0 +1,35 @@
+post:
+  operationId: checkInvitationStatus
+  tags:
+    - Invitations
+  summary: CheckInvitationStatus
+  description: Check the status of an invitation, whether it exits and is valid or not.
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          type: object
+          required:
+            - auth0ClientId
+            - auth0Domain
+            - invitationId
+            - recaptchaToken
+          properties:
+            auth0ClientId:
+              description: the id of the client
+              type: string
+            auth0Domain:
+              description: the domain of the client
+              type: string
+            invitationId:
+              description: the id of the invitation
+              type: string
+            recaptchaToken:
+              description: the user's reCAPTCHA token
+              type: string
+  responses:
+    200:
+      description: invitation exists and is valid and can be used by user
+    400:
+      description: invitation not valid
```

Comment:
Error bodies?

Reply:
No error bodies. Is there info you're expecting here? This is a public API, so I think the less stuff we return the better.

Reply:
Oh, you mean those `ApiError` objects? I can put in a very generic one here.

Reply:
Yeah. Since we concluded that we would do just one POST with the data, the error body needs describe the error for anything that is wrong with submission.

Reply:
Sorry, I might be out-of-date here. What did we conclude and what are we POST-ing here?

Reply:
Thought, for potential subjects: invitationCode , zip, and recaptcha user response.
I believe you were talking about reusing the route for CRs so in that case recaptcha would not be required (though they obviously would have to be authenticated already and have CR role).
Then question would be how you handle ZIP code check for CRs. The simplest approach would be to include the warning message in a response body. Alternative is to have client check it separately. I think the first approach is simpler, but if you feel strongly about leaving up to client to check, I am good with that too.

Reply:
I was thinking zip code will be a separate API. Either way, I think that is outside scope of this PR, and I will like to address it separate from this PR. Is that okay?

Reply:
I still think one remaining issue here is that this API is only for GUI that follows the rules; if I have a valid invitation code I could choose to bypass this whole interaction and go directly to register in Auth0. We might be missing something like a short-lived authorization code that would tell /register route that I was given a green light to register.